src/pkg/build/build.go

```diff
@@ -290,22 +291,28 @@ func (b *Build) runBuildEngine() error {
 	}
 
 	sylog.Debugf("Starting build engine")
+
+	ociConfig := &oci.Config{}
+	generator := generate.Generator{Config: &ociConfig.Spec}
+
+	// Copy and cache environment
+	environment := os.Environ()
+
+	// Clean environment
+	env.SetContainerEnv(&generator, environment, true, "")
```

Comment:
We need to move this from here, in order to respect `%setup` section, which should be for stuff that happens on the host before build.
test it with 
```
BootStrap: busybox
MirrorURL: https://www.busybox.net/downloads/binaries/1.26.1-defconfig-multiarch/busybox-x86_64

%setup
    env
    env > /tmp/pre_build_env

%post
    echo "Hello from inside the container"
    echo $PWD
    env > /tmp/build_env

%runscript
    echo "Running command: $*"
    exec "$@"
```